smart-contracts/wasm-chain-integration/src/v1/trie/low_level.rs

```diff
@@ -450,36 +450,24 @@ impl<V> CachedRef<V> {
         }
     }
 
-    /// If the value is in memory, set it to cached with the given key.
+    /// If the value is in memory, uncache it with the given key.
     /// Otherwise do nothing. This of course has the precondition that the key
     /// stores the value in the relevant backing store. Internal use only.
-    fn cache_with(&mut self, reference: Reference) {
+    fn uncache(&mut self, reference: Reference) {
         if let CachedRef::Memory {
             ..
         } = self
         {
-            let value = std::mem::replace(self, CachedRef::Disk {
+            *self = CachedRef::Disk {
                 reference,
-            });
-            if let CachedRef::Memory {
-                value,
-            } = value
-            {
-                *self = CachedRef::Cached {
-                    value,
-                    reference,
-                };
-            } else {
-                // this is unreachable since we hold a mutable reference to the cached value
-                // and we know the value was a purely in-memory one
-                unsafe { std::hint::unreachable_unchecked() }
-            }
+            };
         }
     }
```

Comment:
Should you not also uncache this if it is "cached"? And set it to the Disk variant?

Reply:
Yes, that is correct. 👍